File: 01-week-1--starter-algorithms/01-day-2--find-first-duplicate/javascript/find_first_duplicate.js
function findFirstDuplicate(arr) {
  // type your code here
  let dupe = ''
  for (i = 0; i < arr.length; i++) {
    if ( arr[i] == arr[i+1]) {
      dupe = arr[i]
    } }
    if (dupe.length == 0 ) {
      dupe = -1
    }
    return dupe
  

}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 3");
  console.log("=>", findFirstDuplicate([2, 1, 3, 3, 2]));

  console.log("");

  console.log("Expecting: -1");
  console.log("=>", findFirstDuplicate([1, 2, 3, 4]));
}

module.exports = findFirstDuplicate;

// Please add your pseudocode to this file
// And a written explanation of your solution
// create an empty array
// iterate over given array
// if the element at a given index is the same as the element at the following index, add it to the new array
// if the new array is still empty, add the integer -1 to the empty array
// return the new array

File: 01-week-1--starter-algorithms/01-day-2--find-first-duplicate/ruby/find_first_duplicate.rb
def find_first_duplicate(arr)
  return -1 if arr.empty?
  hash = Hash.new(0)
  arr.each do |el|
    hash[el] = hash[el] += 1
    return el if hash[el] > 1
  end
  return -1
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 3"
  puts "=>", find_first_duplicate([2, 1, 3, 3, 2])

  puts "Expecting: -1"
  puts "=>", find_first_duplicate([1, 2, 3, 4])

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

# return -1 if the array is empty
# create a new hash to track the values of the array with a null value of 0
# add each arr element to the hash with their own keys. each key's value is their frequency 
# if any arr element's key in the hash has a value greater than 1, return that arr element
# if no duplicates are found at the end of the iteration, return -1
